Trakt/Api/TraktApi.cs

```diff
@@ -923,46 +835,131 @@ private async Task<TraktSyncResponse> SendEpisodePlaystateUpdatesInternalAsync(I
             }
         }
 
-        public async Task RefreshUserAuth(TraktUser traktUser)
+        public string AuthorizeDevice(TraktUser traktUser)
         {
-            var data = new TraktUserTokenRequest
+            var deviceCodeRequest = new
             {
-                client_id = TraktUris.Id,
-                client_secret = TraktUris.Secret,
-                redirect_uri = "urn:ietf:wg:oauth:2.0:oob"
+                client_id = TraktUris.ClientId
             };
-
-            if (!string.IsNullOrWhiteSpace(traktUser.PIN))
+            
+            TraktDeviceCode deviceCode;
+            using (var response = PostToTrakt(TraktUris.DeviceCode, deviceCodeRequest, null))
             {
-                data.code = traktUser.PIN;
-                data.grant_type = "authorization_code";
+                deviceCode = _jsonSerializer.DeserializeFromStream<TraktDeviceCode>(response.Result);
             }
-            else if (!string.IsNullOrWhiteSpace(traktUser.RefreshToken))
+
+            // Start polling in the background
+            Plugin.Instance.PollingTasks[traktUser.LinkedMbUserId] = Task.Run(() => PollForAccessToken(deviceCode, traktUser));
```

Comment:
what happens if there's already a task from same user?

Reply:
I guess there would be two then. Not sure how to handle that. Cancel the existing task?

Reply:
I think check and log. Maybe yes, cancel old one and replace with new if possible.

Reply:
I will have to save the cancellation token to do that I think (I'm not too familiar with tasks)